### PR TITLE
[DRAFT][luci/pass] Support a new pattern in FuseGeluPass

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR supports a new pattern in FuseGeluPass.

for issue:  #11002

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseGeluPass.test.cpp [hunk 1]
```diff
@@ -125,6 +125,32 @@ class FuseGeluTestGraph : public TestIOGraph, public GeluGraphlet
   }
 };
 
+class FuseGeluTestGraph2 : public TestIOGraph, public GeluGraphlet
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseGeluPass.test.cpp:128]: plz separate rename to FuseGeluTestGraph1 and adding this class.

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 2]
```diff
@@ -30,9 +30,81 @@
...
 {
 
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
 // Float comparison
 bool same(float a, float b) { return fabs(a - b) < 1e-5; }
 
+class GeluPatternBase
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseGeluPass.cpp:40]: please split refactoring of existing class to separate PR.

Comment (jinevening) [inline compiler/luci/pass/src/FuseGeluPass.cpp:40]: Please add virtual destructor for parent class.

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 3]
```diff
@@ -62,92 +134,93 @@ bool same(float a, float b) { return fabs(a - b) < 1e-5; }
...
+  bool matched() override;
+};
 
-  // check _const_sqrt condition
-  CHECK_OR_FALSE(_const_sqrt->dtype() == loco::DataType::FLOAT32);
-  CHECK_OR_FALSE(_const_sqrt->size<loco::DataType::FLOAT32>() == 1);
-  CHECK_OR_FALSE(::same(_const_sqrt->at<loco::DataType::FLOAT32>(0), sqrtf(0.5f)));
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
```
Comment (jinevening) [inline compiler/luci/pass/src/FuseGeluPass.cpp:204]: Seems duplicated

File: compiler/luci/pass/src/FuseGeluPass.test.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.test.cpp:128]: I will convert this PR, as a draft, and create PR with renaming and than PR with adding new class

File: compiler/luci/pass/src/FuseGeluPass.cpp (hunk 2 above)
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp:40]: Done

File: compiler/luci/pass/src/FuseGeluPass.cpp (hunk 3 above)
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp]: Done

**Comment by BalyshevArtem** [comment]
done, close it
